src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlExceptionActionProcessor.cs

```diff
@@ -29,6 +31,10 @@ public Task Execute(TRequest request, TException exception, CancellationToken ca
                 {
                     throw new MethodNotAllowedException(Core.Resources.ResourceCreationNotAllowed);
                 }
+                else if (sqlException.Number == QueryProcessorNoQueryPlan)
```

Comment:
Can the QueryProcessorNoQueryPlan number be added to SqlErrorCodes?

Reply:
Yes it can. Doing so would just mean waiting for the shared health to get updated and fhir server to pull that in.

Reply:
Oh, that is in Shared Health. Right...
Lets keep it here then for now and make a PR in Shared Health to add that value. We can update it here latter when we pull the new package.

Reply:
I created a PR in shared health.